xcell/mappers/mapper_2MPZ.py

```diff
@@ -0,0 +1,152 @@
+from .mapper_base import MapperBase
+from .utils import get_map_from_points, get_DIR_Nz
+from astropy.io import fits
+from astropy.table import Table
+import numpy as np
+import healpy as hp
+import os
+
+
+class Mapper2MPZ(MapperBase):
+    def __init__(self, config):
+        """
+        config - dict
+          {'data_catalog': 'Legacy_Survey_BASS-MZLS_galaxies-selection.fits',
+           'mask': 'mask.fits',
+           'z_edges': [0, 0.5],
+           'path_rerun': '.',
+           'n_jk_dir': 100,
+           'mask_name': 'mask_2MPZ'}
+        """
+        self._get_defaults(config)
+        self.z_edges = config.get('z_edges', [0, 0.5])
+
+        self.cat_data = None
+        self.npix = hp.nside2npix(self.nside)
+
+        # Angular mask
+        self.dndz = None
+        self.mask = None
+        self.dndz = None
+        self.delta_map = None
+        self.nl_coupled = None
+        self.mask = None
+
+    def get_catalog(self):
+        if self.cat_data is None:
+            file_data = self.config['data_catalog']
+            if not os.path.isfile(file_data):
+                raise ValueError(f"File {file_data} not found")
+            with fits.open(file_data) as f:
+                self.cat_data = Table.read(f)
+            self.cat_data = self._bin_z(self.cat_data)
+            self.cat_data = self._mask_catalog(self.cat_data)
+
+        return self.cat_data
+
+    def _check_rerun_file_exists(self, fname):
+        path_i = self.config.get('path_rerun', None)
+        if path_i is None:
+            return False, None
+        else:
+            fname_full = os.path.join(path_i, fname)
+            return os.path.isfile(fname_full), fname_full
+
+    def _mask_catalog(self, cat):
+        self.mask = self.get_mask()
+        ipix = hp.ang2pix(self.nside, cat['SUPRA'],
+                          cat['SUPDEC'], lonlat=True)
+        # Mask is binary, so 0.1 or 0.00001 doesn't really matter.
+        return cat[self.mask[ipix] > 0.1]
+
+    def _bin_z(self, cat):
+        return cat[(cat['ZPHOTO'] >= self.z_edges[0]) &
```

Comment:
I don't know if it matters but in KiDS we used the z_min < z <= z_max convention

Reply:
Done